src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/UserAgentTest.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.signalr;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class UserAgentTest {
+
+    private static Stream<Arguments> OperatingSystems() {
+        return Stream.of(
+                Arguments.of("Windows XP", "Windows NT"),
+                Arguments.of("wInDoWs 95", "Windows NT"),
+                Arguments.of("Macintosh", "Mac"),
+                Arguments.of("Linux", "Linux"),
+                Arguments.of("unix", "Linux"),
+                Arguments.of("", ""),
+                Arguments.of("1234", ""));
+    }
+
+    private static Stream<Arguments> Versions() {
+        return Stream.of(
+                Arguments.of("1.0.0", "1.0.0"),
```

Comment:
The short version should be just two segments: `1.0` or `3.1`. The patch version would be in the detailed version. The idea is that you can group on the short version and see all the clients with the same feature set.